src/cli/commands/global.js

```diff
@@ -90,11 +90,15 @@ async function getGlobalPrefix(config: Config, flags: Object): Promise<string> {
   } else {
     prefix = POSIX_GLOBAL_PREFIX;
   }
+
+  const binFolder = path.join(prefix, 'bin');
   try {
-    await fs.access(path.join(prefix, 'bin'), (nativeFs.constants || nativeFs).W_OK);
+    await fs.access(binFolder, (nativeFs.constants || nativeFs).W_OK);
```

Comment:
You should be using the `constants` exposed by `util/fs`, which already do the fallback if required. Also, shouldn't we also test for `R_OK` and `X_OK`?